### ISSUE TITLE
Upgrade Apache Commons Compress to 1.15

### ISSUE THREAD

**Issue by plamentotev** [codehaus-plexus/plexus-archiver#71]
It brings some fixes and most importantly should fix #57

**Comment by plamentotev** [comment]
After the upgrade there is a test failing so I'll postpone the upgrade until the issue is found and resolved.

**Comment by michael-o** [comment]
Are you having the same null character issue? Compress 1.15 seems to be broken for me.

**Comment by plamentotev** [comment]
I think we have the same issue in mind. `ZipArchiverTest#testSymlinkArchivedFileSet` fails because the symlinks in the resulting archive does have several `null` characters at the end. Do you encounter the same issue?

I found the root cause. In `AbstractZipArchiver#zipFile` there is the following code:

```java
            if ( ze.isUnixSymlink() )
            {
                ZipEncoding enc = ZipEncodingHelper.getZipEncoding( getEncoding() );
                final byte[] bytes = enc.encode( symlinkDestination ).array();
                payload = new ByteArrayInputStream( bytes );
                zOut.addArchiveEntry( ze, createInputStreamSupplier( payload ), true );
            }
```
The offender is `enc.encode( symlinkDestination ).array();`. Accordion the `ZipEncoding#encode` Javadocs the returned "...byte buffer is positioned at the beginning of the encoded result, the byte buffer has a backing array and the limit of the byte buffer points to the end of the encoded result." We take the whole backing array. We should take it's content up to the limit instead. Commons Compress 1.15 returns different `ZipEncoding` implementation when `ZipEncodingHelper.getZipEncoding( getEncoding() )` is called and that is why the bug manifests with the update.

I'll submit fix tomorrow after I double check that we don't have the same bug somewhere else in the code.

**Comment by michael-o** [comment]
Correct, I see several null chars in the debugger.

**Comment by michael-o** [comment]
@plamentotev Are you talking about `Buffer#limit()`?

**Comment by plamentotev** [comment]
Yes. I've made a merge request with the fix and the upgrade - #72
